### PR TITLE
Fixes #11864 - introduce remote execution status

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by stbenjam** [comment]
Tests run against 1.9, so this would need to be compatible with both.

**Comment by ares** [comment]
I thought it was (https://github.com/theforeman/foreman_remote_execution/pull/42/files#diff-24c9f52e5c8d0dc8ba4b5d2b7ec5de44R68) but I added host extensions later, will wrap that in condition as well.

File: app/models/concerns/foreman_remote_execution/host_extensions.rb [hunk 1]
```diff
@@ -4,16 +4,27 @@ module HostExtensions
...
-      end
+    def execution_status_label(options = {})
+      @execution_status_label ||= get_status(HostStatus::ExecutionStatus).to_label(options)
     end
+
+    # module ClassMethods
+    #   # create or overwrite class methods...
+    #   def class_method_name
+    #   end
+    # end
```
Comment (stbenjam) [inline app/models/concerns/foreman_remote_execution/host_extensions.rb]: Also please remove this boilerplate

**Comment by ares** [comment]
OK, I think this is a candidate for 1.10 bump, I'll remove not working 1.9 support. Let's keep it open until iteration 2 is done. The issue is that the preloader (in production env) tries to load the execution status which can't find its superclass.

**Comment by stbenjam** [comment]
I think you could wrap the entire class in a check, something like this: https://github.com/theforeman/foreman_salt/blob/4d3af2f4d4b4f4cf68a267a4b8fc389f47e9b9c4/app/lib/actions/foreman_salt/report_import.rb

Up to you though

**Comment by ares** [comment]
yeah, but I think this falls into unnecessary hack category :-)

**Comment by ares** [comment]
@stbenjam could you please change the tests to run against 1.10? This PR is no longer blocked.

File: app/models/host_status/execution_status.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+module HostStatus
```
Comment (ares) [inline app/models/host_status/execution_status.rb]: this should be also under `ForemanOpenscap` module and hence in app/models/foreman_openscap/host_status (host status might not be needed, but leaving this up to you)

**Comment by stbenjam** [comment]
Develop or 1.10? https://github.com/theforeman/foreman-infra/pull/191

**Comment by ares** [comment]
I think 1.10 would be better, but please wait until new release, we realized we'll have to do 0.0.8 that will be still 1.9 compatible (a lot of issues with 0.0.7)

**Comment by stbenjam** [comment]
[test]

**Comment by stbenjam** [comment]
Maybe try what I have here: https://github.com/theforeman/foreman_remote_execution/pull/55/files#diff-24c9f52e5c8d0dc8ba4b5d2b7ec5de44R107

I don't think we need to extend anything but Host::Managed, even unmanaged hosts are still Host::Managed with managed=false.

**Comment by ares** [comment]
I removed Host::Base, but I think we should extend all other descendants (I know only about Host::Discovery but there might be others in future).

**Comment by stbenjam** [comment]
Ok, we might need a separate managed extensions then if we extend managed methods only (like params in my other PR).  Or if aspects is happening anytime soon it can provide a way for aspect parameters.

Anyway, looks like just a typo now.

**Comment by ares** [comment]
[test]

**Comment by stbenjam** [comment]
I bet if you were to change engine.rb to only extend Host::Managed, tests would pass

**Comment by ares** [comment]
should be green after rebasing on top of https://github.com/theforeman/foreman_remote_execution/pull/55 or when https://github.com/theforeman/foreman/pull/2868 gets in

**Comment by stbenjam** [comment]
@ares Needs rebase due to #55

File: app/models/host_status/execution_status.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  def self.status_name
+    N_('Execution')
+  end
+
+  def to_label(options = {})
+    case to_status(options)
+      when OK
+        N_('Last execution succeeded')
```
Comment (stbenjam) [inline app/models/host_status/execution_status.rb]: Sorry one last thought, a host with no executions will have a status that says `Last execution succeeded`, would that be confusing?

Does it make sense to have a different status for that? Like "No executions" that's rated as a Global::OK?

Comment (ares) [inline app/models/host_status/execution_status.rb]: You're right but two different values would result in different numbers so when you search for hosts that does not have any failed execution you'd have to search for `OK or NONE`. So how about changing just to wording to "Last execution succeeded or nothing executed yet"?

Comment (ares) [inline app/models/host_status/execution_status.rb]: or maybe the status should be relevant only if there was some execution?

Comment (stbenjam) [inline app/models/host_status/execution_status.rb]: That description is really long :open_mouth:  Unless you can think of a way to phrase it in a shorter way, I think I prefer making it relevant or not.

**Comment by ares** [comment]
So I changed it to be irrelevant until there's at least one task. In case another status calls refresh and this task becomes relevant (there's at least one task but none is stopped) we print different message while still remaining OK.

**Comment by stbenjam** [comment]
Works great ! ACK

**Comment by ares** [comment]
Thanks, merging

**Pull request merged**